api_v3/services/MediaService.php

```diff
@@ -1090,10 +1090,19 @@ protected function prepareEntryForInsert(KalturaBaseEntry $entry, entry $dbEntry
 		$entry->validatePropertyNotNull("mediaType");
 
 		$conversionQuality = $this->getConversionQuality($entry);
-		if (!is_null($conversionQuality))
+		if (!is_null($conversionQuality)) {
 			$entry->conversionQuality = $conversionQuality;
+			if (!$entry->conversionProfileId) {
+				$entry->conversionProfileId = $entry->conversionQuality;
+			}
+		}
+
+		if ($dbEntry == null){
+			$dbEntry = $this->duplicateTemplateEntry($entry->conversionProfileId);
```

Comment:
The name conversion-profile is confusing, it should be named ingestion-profile, it contains template-entry-id, storage-profile-id, and specifically to document-entries it's very relevant as it's used just like in media-entries to define the transcoded flavors.